runners/portability/java/src/main/java/org/apache/beam/runners/portability/ExternalWorkerService.java

```diff
@@ -130,15 +131,13 @@ public void close() {}
    *       up-front and matches the running job. See {@link PipelineOptions} for further details.
    * </ul>
    */
-  public static void main(String[] args) throws Exception {
-    main(System::getenv);
-  }
-
-  public static void main(Function<String, String> environmentVarGetter) throws Exception {
+  public static void main(String[] args) {
     LOG.info("Starting external worker service");
-    LOG.info("Pipeline options {}", environmentVarGetter.apply(PIPELINE_OPTIONS_ENV_VAR));
-    PipelineOptions options =
-        PipelineOptionsTranslation.fromJson(environmentVarGetter.apply(PIPELINE_OPTIONS_ENV_VAR));
+    final String optionsEnv =
+        checkArgumentNotNull(
+            System.getenv(PIPELINE_OPTIONS_ENV_VAR), "No pipeline options provided.");
```

Comment:
Make the error message actionable. Tell the user how to set PIPELINE_OPTIONS.

Reply:
What about 
```
"No pipeline options provided in environment variables " + PIPELINE_OPTIONS_ENV_VAR
```

Reply:
Sounds good.

Reply:
Updated